pkg/driver/agent.go

```diff
@@ -171,7 +172,16 @@ func (ns *node) NodeGetInfo(
 	req *csi.NodeGetInfoRequest,
 ) (*csi.NodeGetInfoResponse, error) {
 
-	topology := map[string]string{zfs.ZFSTopologyKey: ns.driver.config.NodeID}
+	node, err := k8sapi.GetNode(ns.driver.config.NodeID)
+	if err != nil {
+		logrus.Errorf("failed to get the node %s", ns.driver.config.NodeID)
+		return nil, err
+	}
+
+	// support all the keys that node has
+	topology := node.Labels
+	topology[zfs.ZFSTopologyKey] = ns.driver.config.NodeID
```

Comment:
An comment here with example labels will be good.

Reply:
added comment.